dev-utils/visualise-ui-hierarchy/Makefile

```diff
@@ -0,0 +1,46 @@
+CWD := $(shell pwd)
+POLYPOD_FEATURES_PATH := $(CWD)/../../features
+RCH := node $(CWD)/react-component-hierarchy/bin/rch.js -e
+
+.PHONY: all
+all: polyExplorer.svg facebookImport.svg
+
+.PHONY: rebuild
+rebuild: clean-files all
+
+%.svg: %.dot
+	fdp -T svg -o $@ $<
+
+.INTERMEDIATE: %.dot
+%.dot: %.txt
+	./tree2dot < $< > $@
+
+.INTERMEDIATE: polyExplorer.txt
+polyExplorer.txt: ensure-rch
+	cd $(POLYPOD_FEATURES_PATH)/polyExplorer; \
+		$(RCH) src/polyExplorer.jsx > $@; \
+		mv $@ $(CWD)
+
+.INTERMEDIATE: facebookImport.txt
+facebookImport.txt: ensure-rch
```

Comment:
You can try and use a [*canned recipe*](https://www.gnu.org/software/make/manual/html_node/Canned-Recipes.html) to DRY here.

Reply:
Good point, that was nasty. Done!